sway/input/cursor.c

```diff
@@ -19,16 +20,12 @@ static void cursor_update_position(struct sway_cursor *cursor) {
 	cursor->y = y;
 }
 
-static void cursor_send_pointer_motion(struct sway_cursor *cursor,
-		uint32_t time) {
-	struct wlr_seat *seat = cursor->seat->wlr_seat;
-	struct wlr_surface *surface = NULL;
-	double sx, sy;
-
+static struct sway_container *cursor_at(struct sway_cursor *cursor,
+		struct wlr_surface **surface, double *sx, double *sy) {
 	// check for unmanaged views first
+	struct wl_list *unmanaged = &root_container.sway_root->unmanaged_views;
 	struct sway_view *view;
-	wl_list_for_each_reverse(view, &root_container.sway_root->unmanaged_views,
-		unmanaged_view_link) {
+	wl_list_for_each_reverse(view, unmanaged, unmanaged_view_link) {
```

Comment:
I think the logic to find unmanaged views should be within the `container_at` function. They are on top of everything, so it should probably run no matter what the parent is.

Reply:
Problem is, `container_at` takes a container to search views in. Unmanaged views are always outside it.